Lib/logging/__init__.py

```diff
@@ -426,19 +429,44 @@ class PercentStyle(object):
     def __init__(self, fmt):
         self._fmt = fmt or self.default_format
 
+    @contextmanager
+    def _verifyFields(self):
+        """
+        Verify if fields passed into XXXStyle.
+        """
+        try:
+            yield
+        except KeyError as e:
+            raise ValueError("Invalid formatting field %s" % e)
+
     def usesTime(self):
         return self._fmt.find(self.asctime_search) >= 0
 
+    def validate(self):
+        self._validateFormat(r'(\%\([^%()]+\)[\s\d-]*[\w]+)')
```

Comment:
Hmm, this does seem to match the example you have given. let me try and redo this.

Reply:
I think I have gotten quite a few things wrong in this pattern. I thought something like this would be valid`%(asctime)- 15s`, but looking at the specifications, I definitely got it wrong